src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningConditionMapper.java

```diff
@@ -7,55 +7,75 @@
+import com.salesforce.dynamodbv2.mt.mappers.index.DynamoSecondaryIndex;
+import com.salesforce.dynamodbv2.mt.mappers.metadata.DynamoTableDescription;
 import com.salesforce.dynamodbv2.mt.mappers.metadata.PrimaryKey;
-import com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl.HashPartitioningKeyMapper.PhysicalRangeKeyBytesConverter;
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl.HashPartitioningKeyMapper.HashPartitioningKeyBytesConverter;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
-import java.util.stream.Collectors;
-import org.antlr.v4.runtime.BailErrorStrategy;
-import org.antlr.v4.runtime.BaseErrorListener;
-import org.antlr.v4.runtime.CharStreams;
-import org.antlr.v4.runtime.CommonTokenStream;
-import org.antlr.v4.runtime.RecognitionException;
-import org.antlr.v4.runtime.Recognizer;
 
-class HashPartitioningConditionMapper implements ConditionMapper {
+class HashPartitioningConditionMapper extends AbstractConditionMapper {
 
     private static final byte[] EMPTY_BYTE_ARRAY = new byte[] {};
 
     private final HashPartitioningKeyMapper keyMapper;
+    private final Multimap<String, String> secondaryIndexFieldPartners;
 
-    HashPartitioningConditionMapper(HashPartitioningKeyMapper keyMapper) {
+    HashPartitioningConditionMapper(DynamoTableDescription virtualTable,
+                                    HashPartitioningItemMapper itemMapper,
+                                    HashPartitioningKeyMapper keyMapper) {
+        super(virtualTable, itemMapper);
         this.keyMapper = keyMapper;
+        this.secondaryIndexFieldPartners = getSecondaryIndexFieldPartners(virtualTable);
+    }
+
+    /**
+     * Returns a Multimap of field pairings in composite secondary indexes, so we can easily enforce that a field can
+     * be updated only if for each composite secondary index that the field is part of, the other index field is also
+     * being updated.
+     */
+    private static Multimap<String, String> getSecondaryIndexFieldPartners(DynamoTableDescription virtualTable) {
+        Multimap<String, String> fieldPartners = LinkedListMultimap.create();
+        for (DynamoSecondaryIndex secondaryIndex : virtualTable.getSis()) {
+            if (secondaryIndex.getPrimaryKey().getRangeKey().isPresent()) {
+                String hashKey = secondaryIndex.getPrimaryKey().getHashKey();
+                String rangeKey = secondaryIndex.getPrimaryKey().getRangeKey().get();
+                fieldPartners.put(hashKey, rangeKey);
+                fieldPartners.put(rangeKey, hashKey);
+            }
+        }
+        return fieldPartners;
     }
 
     @Override
-    public void applyForUpdate(UpdateItemRequest updateItemRequest) {
-        // TODO
+    protected void validateFieldsCanBeUpdated(Set<String> allSetFields) {
+        super.validateFieldsCanBeUpdated(allSetFields);
+
```

Comment:
This is where we enforce that a user can't update just one but not both of the fields in a composite secondary index